File: server/src/knowledge/ingestors/docxIngestor.js
import { BaseIngestor } from '../baseIngestor.js';

// Needs a real docx parser (e.g. `mammoth`) — not installed yet.
export class DocxIngestor extends BaseIngestor {
  constructor() {
    super('document');
  }

  async ingest(_source) {
    throw new Error('DocxIngestor not implemented — add `mammoth` package for text extraction');
  }
}

File: server/app/knowledge/ingestors/docx_ingestor.py
from ..base_ingestor import BaseIngestor


class DocxIngestor(BaseIngestor):
    def __init__(self):
        super().__init__("document")

    async def ingest(self, source: dict) -> list[dict]:
        raise NotImplementedError("DocxIngestor not implemented — see rag.parse_document for the wired docx path")
